src/error.jl

```diff
@@ -1,41 +1,42 @@
-using SumTypes: @sum_type
-
 export get_error_code, get_error_message
 
-# See https://www.mcobject.com/docs/Content/Programming/C/Return_Codes.htm or
-# https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html
-"Represent various return codes from the Spglib library."
-@sum_type SpglibReturnCode begin
-    SUCCESS
-    SPACEGROUP_SEARCH_FAILED
-    CELL_STANDARDIZATION_FAILED
-    SYMMETRY_OPERATION_SEARCH_FAILED
-    ATOMS_TOO_CLOSE
-    POINTGROUP_NOT_FOUND
-    NIGGLI_FAILED
-    DELAUNAY_FAILED
-    ARRAY_SIZE_SHORTAGE
-    NONE
-end
+# Mapping of the `SpglibError` C enum variants to integer values.
+const SPGLIB_SUCCESS = 0
+const SPGERR_SPACEGROUP_SEARCH_FAILED = 1
+const SPGERR_CELL_STANDARDIZATION_FAILED = 2
+const SPGERR_SYMMETRY_OPERATION_SEARCH_FAILED = 3
+const SPGERR_ATOMS_TOO_CLOSE = 4
+const SPGERR_POINTGROUP_NOT_FOUND = 5
+const SPGERR_NIGGLI_FAILED = 6
+const SPGERR_DELAUNAY_FAILED = 7
+const SPGERR_ARRAY_SIZE_SHORTAGE = 8
+const SPGERR_NONE = 9
 
 struct SpglibError <: Exception
     msg::String
 end
 
+# The return value for the C function `spg_get_error_code` is a C enum called
+# `SpglibError`. The C compiler guarantees that enums are internally stored as
+# type `int`, which Julia represents as `Cint`. The size of `int` is
+# "implementation dependent". Frequently 4 or 8 bytes, but this depends on
+# architecture and compiler details: https://stackoverflow.com/a/366033/500314.
+# Directly casting this Cint into an @enum or SumTypes.jl value would be memory
+# unsafe (see https://github.com/singularitti/Spglib.jl/issues/183).
 """
     get_error_code()
 
 Return an instance of the enumerated type `SpglibReturnCode`.
 """
-get_error_code() = @ccall libsymspg.spg_get_error_code()::SpglibReturnCode
+get_error_code() = @ccall libsymspg.spg_get_error_code()::Cint
```

Comment:
The return type of `Cint` is critical here -- that's what `spg_get_error_code` actually produces.